Backported patch by Robert Scheck <robert@fedoraproject.org> for sendmail 8.14.4, which changes
the (broken) behaviour of FEATURE(`ldap_routing') regarding -T<TMPF> option to the same, that is
used now by version 8.14.5 and later. See https://bugzilla.redhat.com/show_bug.cgi?id=890227 for
more information and details.

--- sendmail-8.14.4/cf/feature/ldap_routing.m4			2009-06-26 23:11:08.000000000 +0200
+++ sendmail-8.14.4/cf/feature/ldap_routing.m4.ldap_routing	2012-01-18 23:27:06.000000000 +0100
@@ -1,6 +1,6 @@
 divert(-1)
 #
-# Copyright (c) 1999-2002, 2004, 2007, 2009 Sendmail, Inc. and its suppliers.
+# Copyright (c) 1999-2002, 2004, 2007 Sendmail, Inc. and its suppliers.
 #	All rights reserved.
 #
 # By using this file, you agree to the terms and conditions set
@@ -10,7 +10,7 @@
 #
 
 divert(0)
-VERSIONID(`$Id: ldap_routing.m4,v 8.17 2009/06/26 21:11:08 ca Exp $')
+VERSIONID(`$Id: ldap_routing.m4,v 8.20 2012/01/18 22:27:06 ca Exp $')
 divert(-1)
 
 # Check first two arguments.  If they aren't set, may need to warn in proto.m4
@@ -35,40 +35,12 @@
        _ARG6_, `tempfail', `define(`_LDAP_ROUTE_MAPTEMP_', `_TEMPFAIL_')',
        _ARG6_, `queue', `define(`_LDAP_ROUTE_MAPTEMP_', `_QUEUE_')')
 
-define(`_ATMPF_', `<TMPF>')dnl
-dnl check whether arg contains -T`'_ATMPF_
-dnl unless it is a sequence map or just LDAP
-dnl note: this does not work if ARG1 begins with space(s), however, as
-dnl we issue a warning, hopefully the user will fix it...
-ifelse(defn(`_ARG1_'), `', `',
-  defn(`_ARG1_'), `LDAP', `',
-  `ifelse(index(_ARG1_, `sequence '), `0', `',
-    `ifelse(index(_ARG1_, _ATMPF_), `-1',
-      `errprint(`*** WARNING: missing -T'_ATMPF_` in first argument of FEATURE(`ldap_routing')
-')
-      define(`_ABP_', index(_ARG1_, ` '))
-      define(`_NARG1_', `substr(_ARG1_, 0, _ABP_) -T'_ATMPF_` substr(_ARG1_, _ABP_)')
-      ')
-    ')
-  ')
-ifelse(defn(`_ARG2_'), `', `',
-  defn(`_ARG2_'), `LDAP', `',
-  `ifelse(index(_ARG2_, `sequence '), `0', `',
-    `ifelse(index(_ARG2_, _ATMPF_), `-1',
-      `errprint(`*** WARNING: missing -T'_ATMPF_` in second argument of FEATURE(`ldap_routing')
-')
-      define(`_ABP_', index(_ARG2_, ` '))
-      define(`_NARG2_', `substr(_ARG2_, 0, _ABP_) -T'_ATMPF_` substr(_ARG2_, _ABP_)')
-      ')
-    ')
-  ')
-
 LOCAL_CONFIG
 # LDAP routing maps
 Kldapmh ifelse(len(X`'_ARG1_), `1',
 	       `ldap -1 -T<TMPF> -v mailHost -k (&(objectClass=inetLocalMailRecipient)(mailLocalAddress=%0))',
-	       defn(`_NARG1_'), `', `_ARG1_', `_NARG1_')
+	       `_ARG1_')
 
 Kldapmra ifelse(len(X`'_ARG2_), `1',
 		`ldap -1 -T<TMPF> -v mailRoutingAddress -k (&(objectClass=inetLocalMailRecipient)(mailLocalAddress=%0))',
-		defn(`_NARG2_'), `', `_ARG2_', `_NARG2_')
+		`_ARG2_')
